src/interfaces/protocols/types.ts

```diff
@@ -10,17 +10,18 @@ export type ProtocolsConfigureDescriptor = {
 };
 
 export type ProtocolDefinition = {
-  labels: {
-    [key: string]: {
-      schema: string,
-      dataFormats?: string[],
-    };
-  };
+  recordTypes: ProtocolRecordType[];
   records: {
     [key: string]: ProtocolRuleSet;
   };
 };
 
+export type ProtocolRecordType = {
```

Comment:
Discussion: frankly I am still not 100% with "Record Type", because "schema" and "type" feels interchangeable in common programming lingo, it is not the case here, hence the hesitation.

I feel "constraint" is the best word I can find that says what this data is for, but also not 100% happy if that means we end up calling the referencing property name under `records` hierarchy `recordConstraint` instead of `recordType`.

What if we call `recordTypes` -> `recordDefinitions`? And in `records` hierarchy, we just call the referencing property `record`?

Reply:
This works for me. I like `recordDefinitions` better than `recordTypes` now that you mention it. `record` property in the hierarchy also works for me.